src/var.cc

```diff
@@ -85,6 +85,33 @@ void Var::Used(Evaluator* ev, const Symbol& sym) const {
   }
 }
 
+void Var::SetVisibilityPrefix(const std::vector<std::string>& prefixes, const char* name) {
+  std::vector<std::string> current_prefixes = VisibilityPrefix();
+  if (current_prefixes.size() == 0) {
+    visibility_prefix_ = prefixes;
+  } else if (current_prefixes != prefixes) {
+    ERROR("Visibility prefix conflict on variable: %s", name);
+  }
+}
+
+void Var::CheckCurrentReferencingFile(Loc loc, const char* name) {
+  std::vector<std::string> prefixes = VisibilityPrefix();
+  if (prefixes.size() == 0) {
+    return;
+  }
+  std::string referencingFile(loc.filename);
+  bool valid = false;
+  for (const std::string& prefix : prefixes) {
+    if (referencingFile.find(prefix) == 0) {
+      valid = true;
+      break;
+    }
+  }
+  if (!valid) {
+    ERROR("%s is not a valid file to reference variable %s. Line #%d", loc.filename, name, loc.lineno);
```

Comment:
Please print out the valid locations the variable can be in as part of the error message.